validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -222,13 +223,29 @@ public void subscribeToPersistentSubnets(Set<SubnetSubscription> subnetSubscript
   }
 
   @Override
-  public void sendSignedAttestation(final Attestation attestation) {
+  public void sendSignedAttestation(
+      final Attestation attestation, final Optional<Integer> expectedValidatorIndex) {
     attestationManager
         .onAttestation(ValidateableAttestation.fromAttestation(attestation))
         .ifInvalid(
-            reason ->
-                VALIDATOR_LOGGER.producedInvalidAttestation(
-                    attestation.getData().getSlot(), reason));
+            reason -> {
+              int actualValidatorIndex =
+                  get_attesting_indices(
+                          combinedChainDataClient.getHeadStateFromStore().orElseThrow(),
+                          attestation.getData(),
+                          attestation.getAggregation_bits())
+                      .get(0);
+              VALIDATOR_LOGGER.producedInvalidAttestation(
+                  attestation.getData().getSlot(),
+                  actualValidatorIndex,
+                  expectedValidatorIndex,
+                  reason);
+            });
+  }
+
+  @Override
+  public void sendSignedAttestation(final Attestation attestation) {
+    sendSignedAttestation(attestation, Optional.empty());
```

Comment:
Do we need this method?  Could we just require the `validatorIndex` as a second param and drop the `Optional`?

Reply:
I kept it because otherwise, I would need to change a lot of tests, and since we don't normally require this change and only doing it for debugging purposes, I decided to keep this and pass the validator index as optional. I don't have a strong preference though.